examples/webgl_geometry_compression.html

```diff
@@ -0,0 +1,298 @@
+
+				return geom;
+			}
+
+			function generateGeometry() {
+
+				updateGroupGeometry(
+					ballMesh,
+					ballLineSegments,
+					newGeometry(),
+					data);
+			}
+
+			// updateLineSegments
+			function updateLineSegments() {
+				ballLineSegments.visible = data.wireframe;
+			}
+
+			var folder = gui.addFolder('Scene');
+			folder.open();
+			folder.add(data, 'wireframe', false).onChange(updateLineSegments);
+			folder.add(data, 'texture', false).onChange(generateGeometry);
+			folder.add(data, 'detail', 0, 6, 1).onChange(generateGeometry);
+			folder.add(data, 'rotationSpeed', 0, 0.5, 0.1);
+
+			folder = gui.addFolder('Position Compression');
+			folder.open();
+			folder.add(data, 'quantizeEncodePos', false).onChange(generateGeometry);
+
+			folder = gui.addFolder('Normal Compression');
+			folder.open();
+			folder.add(data, 'defaultEncodeNormal', false).onChange(generateGeometry);
+			folder.add(data, 'anglesEncodeNormal', false).onChange(generateGeometry);
+			folder.add(data, 'oct1bytesEncode', false).onChange(generateGeometry);
+			folder.add(data, 'oct2bytesEncode', false).onChange(generateGeometry);
+
+			folder = gui.addFolder('UV Compression');
+			folder.open();
+			folder.add(data, 'defaultEncodeUV', false).onChange(generateGeometry);
+
+			folder = gui.addFolder('Memory Info');
+			folder.open();
+			memoryDisplay = folder.add(data, 'totalGPUMemory', "0 bytes");
+			computeGPUMemory(ballMesh);
+
+			//
+
+			if (statsEnabled) {
+
+				stats = new Stats();
+				container.appendChild(stats.dom);
+
+			}
+
+			window.addEventListener('resize', onWindowResize, false);
+
+		}
+
+		//
+
+		function onWindowResize() {
+
+			renderer.setSize(window.innerWidth, window.innerHeight);
+
+			camera.aspect = window.innerWidth / window.innerHeight;
+			camera.updateProjectionMatrix();
+
+		}
+
+		//
+		function updateLightsPossition() {
+			lights.forEach(light => {
+				var direction = light.position.clone();
+				direction.applyAxisAngle(new THREE.Vector3(1, 1, 0), data.rotationSpeed / 180 * Math.PI);
+				light.position.add(direction.sub(light.position));
+			});
+		}
+
+		//
+
+		function animate() {
+
+			requestAnimationFrame(animate);
+
+			controls.update();
+			updateLightsPossition();
+
+			renderer.render(scene, camera);
+
+			if (statsEnabled) stats.update();
+
+		}
+
+		//
+
+		function updateGroupGeometry(mesh, lineSegments, geometry, data) {
+
+			if (geometry.isGeometry) {
+
+				geometry = new THREE.BufferGeometry().fromGeometry(geometry);
+
+				console.warn('THREE.GeometryBrowser: Converted Geometry to BufferGeometry.');
+
+			}
+
+			lineSegments.geometry.dispose();
+			mesh.geometry.dispose();
+
+			lineSegments.geometry = new THREE.WireframeGeometry(geometry);
+			mesh.geometry = geometry;
+			mesh.material = new THREE.MeshPhongMaterial({ color: 0xffffff, emissive: 0x111111 });
+			mesh.material.map = data.texture ? texture : null;
+
+			var normalEncode = "";
+			if (data.oct1bytesEncode) {
```

Comment:
@LeonYuanYao This should probably be changed to a dropdown in the example GUI since they're all mutually exclusive options. It's a bit confusing at the moment because all of the normal packing options can be checked but it will only use the last one.